component/src/main/java/org/wso2/siddhi/runner/beam/SiddhiAppContainer.java

```diff
@@ -0,0 +1,287 @@
+            this.streamDefinitions.forEach(siddhiApp::append);
+            this.queryDefinitions.forEach(siddhiApp::append);
+            log.debug(siddhiApp.toString());
+            siddhiManager.setExtension("beam:pardo", BeamParDoProcessor.class);
+            siddhiManager.setExtension("beam:groupbykey", BeamGroupByKeyProcessor.class);
+            siddhiManager.setExtension("beam:sink", BeamSinkProcessor.class);
+            this.runtime = siddhiManager.createSiddhiAppRuntime(siddhiApp.toString());
+            this.runtime.start();
+        } catch (SiddhiParserException e) {
+            log.error("There is an error in your SiddhiQL", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    private void generateSiddhiQueryForTransform(AppliedPTransform transform, PCollection keyCollection) {
+        /*
+        If transform is not in HashMap
+         */
+        if (this.transformsMap.get(SiddhiAppContainer.generateTransformName(transform.getFullName())) == null) {
+            /*
+            Add stream definition and to HashMap for given transform
+             */
+            String streamName = SiddhiAppContainer.generateTransformName(transform.getFullName()) + "Stream";
+            String stream = "define stream " + streamName + " (event object);";
+            this.streamDefinitions.add(stream);
+            this.transformsMap.put(SiddhiAppContainer.generateTransformName(transform.getFullName()), transform);
+            this.collectionsMap.put(SiddhiAppContainer.generateTransformName(transform.getFullName()), keyCollection);
+
+            /*
+            Create queries for each transform mapped by output collections
+             */
+            if (transform.getOutputs().isEmpty()) {
+                if (transform.getTransform() instanceof TextIO.Write) {
+                    String sinkType = "text";
+                    String sinkStreamName = "textSinkStream"
+                            + (multiSinkcount == 1 ? "" : String.valueOf(multiSinkcount));
+                    try {
+                        TextIO.TypedWrite textio = ((TextIO.Write) transform.getTransform()).withOutputFilenames();
+                        Class<?> cls = textio.getClass();
+                        Method getFileNamePrefix = cls.getDeclaredMethod("getFilenamePrefix");
+                        getFileNamePrefix.setAccessible(true);
+                        ValueProvider.StaticValueProvider provider =
+                                (ValueProvider.StaticValueProvider) getFileNamePrefix.invoke(textio);
+                        String filePath = provider.get().toString();
+                        this.queryDefinitions.add(generateSinkQuery(sinkType, streamName, sinkStreamName));
+                        this.streamDefinitions.add(generateSinkStream(sinkType, sinkStreamName, filePath));
+                    } catch (NoSuchMethodException | SecurityException exception) {
+                        log.error(exception.getMessage(), exception);
```

Comment:
Shall we add a error message with context. For example error occurred while creating queries for transform mapped by output collections